### PR TITLE
Improve precision for dirchlet sampler

### PR DESCRIPTION
**Pull request by fehiepsi**
Addresses https://github.com/pyro-ppl/numpyro/issues/942

Current tests for `beta` and `dirichlet` should exercise the change.

image

### CODE REVIEW

**Review by fritzo — DISMISSED** [review]

File: test/test_distributions.py [hunk 1]
```diff
@@ -450,6 +450,7 @@ def test_sample_gradient(jax_dist, sp_dist, params):
         "Gamma": ["concentration"],
         "Beta": ["concentration1", "concentration0"],
         "Chi2": ["df"],
+        "Dirichlet": ["concentration"],
```
Comment (fehiepsi) [inline test/test_distributions.py:453]: previously, this test passed for Dirichlet is just an artifact (e.g. due to some specific parameters/random seeds); we should skip it here.

Comment (fritzo) [reply test/test_distributions.py:453]: IIUC the reparametrized gradients for Gamma, Beta, Dirichlet, Chi2, and StudentT are all stochastic in the sense that gradients are correct only in expectation.  You might instead add a weaker test that drew many samples and estimated the Jacobian `d/dtheta E_p(z|theta)[ f(z) ]` for the parameter `theta` and some collection of functions `f` like `[1, z, z**2, z**3, z**4]`.

Comment (fritzo) [reply test/test_distributions.py:453]: See e.g. Figure 2 in https://arxiv.org/pdf/1806.01851.pdf

Comment (fehiepsi) [reply test/test_distributions.py:453]: Thanks, currently we have a TODO for them at test_pathwise_gradient. Let me create a separate issue instead.

**Comment by fehiepsi** [comment]
Thanks for reviewing, @fritzo! I just fixed an import lint issue since your last approval. Could you help me approve this again? Thanks!

**Review by fritzo — APPROVED** [review]

**Pull request merged**